pythondata_cpu_rocket/verilog/update.sh

```diff
@@ -98,60 +118,120 @@ cat >> rocket-chip/src/main/scala/system/Configs.scala <<- "EOT"
 	)
 
 	class WithLitexHextConfig extends Config(
-	  new WithHypervisor ++
-	  new WithBitManip ++ new WithBitManipCrypto ++
-	  new WithCryptoNIST ++ new WithCryptoSM
+	  new WithHypervisor
 	)
 	EOT
 
+# if use firtool old than 1.60, uncomment this
+#sed -i '/"-disable-infer-rw",/d' build.sc
+#sed -i '/"-dedup",/d' build.sc
+
+# TODO: This should be fixed by changing new name in upstream
+sed -i "s/('_')/('|')/" build.sc
+
 # Fit Rocket core models to each LiteX model:
-declare -A CORE_TYPE=(
+declare -r -A CORE_TYPE=(
```

Comment:
making the array "read-only" -- stylistic vs. necessary/meaningful?

Reply:
I think this it is necessary, since we only use this as a mapping, declare `CORE_TYPE` as read-only will avoid mistakes.